eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/ReplaceableTxFunder.scala

```diff
@@ -88,6 +90,35 @@ object ReplaceableTxFunder {
     addSigs(unsignedCommitTx, PlaceHolderPubKey, PlaceHolderPubKey, PlaceHolderSig, PlaceHolderSig)
   }
 
+  /**
+   * The on-chain feerate can be arbitrarily high, but it wouldn't make sense to pay more fees than the amount we're
+   * trying to claim on-chain. We compute how much funds we have at risk and the feerate that matches this amount.
+   */
+  def maxFeerate(txInfo: ReplaceableTransactionWithInputInfo, commitment: FullCommitment): FeeratePerKw = {
+    // We don't want to pay more in fees than the amount at risk in untrimmed pending HTLCs.
+    val maxFee = txInfo match {
+      case tx: HtlcTx => tx.input.txOut.amount
+      case tx: ClaimHtlcTx => tx.input.txOut.amount
+      case _: ClaimLocalAnchorOutputTx =>
+        val htlcBalance = commitment.localCommit.htlcTxsAndRemoteSigs.map(_.htlcTx.input.txOut.amount).sum
+        val mainBalance = commitment.localCommit.spec.toLocal.truncateToSatoshi
```

Comment:
The asymmetry between htlc and main is suspicious, but I understand this is a simplification.

Reply:
We can really discuss that part, because there are two conflicting requirements:

- on one hand, we generally don't want to pay more fees than our total outstanding htlc amount (because those are the funds at risk of being claimed by our peer)
- but when there are no HTLCs or only small ones, we'll also want to get back the funds from our `toLocal` output, because it's otherwise liquidity that is unavailable, but we don't want to waste too much of that amount on fees (so I arbitrarily chose 5%)

I think it's worth thinking about those choices and see if they really make sense or if something else would be better.

Reply:
I think your value makes sense. It's better than what we currently have in any case.